src/mescal/agave_de_v2.clj

```diff
@@ -59,6 +60,31 @@
   [agave app-id]
   (mapv :id (:inputs (.getApp agave app-id))))
 
+(defn list-app-permissions
+  [agave app-ids]
+  (map #(apps/format-app-permissions % (.listAppPermissions agave %)) app-ids))
+
+(defn has-app-permission
+  [agave username app-id required-level]
+  (try+
+    (let [{:keys [write read execute]} (:permission (.getAppPermission agave app-id username))]
+      (case required-level
+        "own"     write
+        "write"   write
+        "read"    read
+        "execute" execute
+        false))
```

Comment:
I agree, I think those edge cases shouldn't arise between users only using the DE, but they might crop up more often for users setting permissions only via Agave.

If the permissions are set via the DE, these values should pretty much reflect DE permission values, since we either remove all Agave permissions for the DE `unshare` action, give Agave `read_execute` permission for the DE `read` permission, or `all` Agave permissions for any other DE permission.

If a user grants only an Agave `write`, `read`, or `execute` permission for their app to another user, then this could cause unexpected behavior for the other user in the DE.

Also, we'll have to let our DE users know that granting `write` access for their Agave app to other users will allow those users to update the permissions themselves (since this translates to DE `own` permissions).